test/jdk/java/nio/channels/DatagramChannel/Loopback.java

```diff
@@ -157,10 +160,15 @@ static void test(ProtocolFamily family, InetAddress group, NetworkInterface ni)
                     } else {
                         sel.selectedKeys().clear();
                         SocketAddress sender = dc.receive(dst);
+                        if (src.mismatch(dst) != -1) {
+                            System.out.println("src: " + src + "not equal to dst: " + dst);
+                            dst.clear();
+                            continue;
```

Comment:
So if this happens we will go back waiting for 3 more seconds. We're not supposed to receive stray datagrams here, so I assume it might be OK to respin with 3 full seconds as this shouldn't happen too often...

Reply:
Okey, I have added Thread.sleep(3000) in while loop.